src/NVMLDevicePool.cpp

```diff
@@ -136,251 +138,170 @@ namespace geopm
 
         nvmlReturn_t nvml_result = nvmlDeviceGetCpuAffinity(m_nvml_device.at(accel_idx), cpu_set_size,
                                                             (unsigned long *)accel_cpuset);
-        if (NVML_SUCCESS != nvml_result) {
-            throw Exception("NVMLDevicePool::" + std::string(__func__) +
-                            ": NVML failed to get CPU Affinity bitmask for accelerator " +
-                            std::to_string(accel_idx) + ".  Error: " + nvmlErrorString(nvml_result),
-                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
-        }
-
+        check_nvml_result(nvml_result, GEOPM_ERROR_RUNTIME, "NVMLDevicePool::" + std::string(__func__) +
+                          ": NVML failed to get CPU Affinity bitmask for accelerator " +
+                          std::to_string(accel_idx) + ".  Error: " + nvmlErrorString(nvml_result));
```

Comment:
`nvmlErrorString(nvml_result)` could probably move inside the function too.  Either way is fine.